pandas/core/dtypes/cast.py

```diff
@@ -707,7 +707,7 @@ def infer_dtype_from_scalar(val, pandas_dtype: bool = False):
     return dtype, val
 
 
-def infer_dtype_from_array(arr, pandas_dtype: bool = False):
+def infer_dtype_from_array(arr, pandas_dtype: bool = False) -> Tuple[DtypeObj, Any]:
```

Comment:
maybe a TODO: that the Any is to ensure a two tuple and not to represent dynamic typing and that further refinement is required.